examples/generate.sh

```diff
@@ -0,0 +1,106 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+# Directories.
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+OUTPUT_DIR=${OUTPUT_DIR:-${SOURCE_DIR}/_out}
+ENVSUBST=${ENVSUBST:-envsubst}
+CLUSTERAWSADM=${CLUSTERAWSADM:-clusterawsadm}
+
+# Cluster name.
+export CLUSTER_NAME="${CLUSTER_NAME:-test1}"
+
+# Machine settings.
+export CONTROL_PLANE_MACHINE_TYPE="${CONTROL_PLANE_MACHINE_TYPE:-t2.medium}"
+export NODE_MACHINE_TYPE="${CONTROL_PLANE_MACHINE_TYPE:-t2.medium}"
+export SSH_KEY_NAME="${SSH_KEY_NAME:-default}"
+
+# Outputs.
+COMPONENTS_CLUSTER_API_GENERATED_FILE=${SOURCE_DIR}/provider-components/provider-components-cluster-api.yaml
+COMPONENTS_KUBEADM_GENERATED_FILE=${SOURCE_DIR}/provider-components/provider-components-kubeadm.yaml
+COMPONENTS_AWS_GENERATED_FILE=${SOURCE_DIR}/provider-components/provider-components-aws.yaml
+
+PROVIDER_COMPONENTS_GENERATED_FILE=${OUTPUT_DIR}/provider-components.yaml
+CLUSTER_GENERATED_FILE=${OUTPUT_DIR}/cluster.yaml
+CONTROLPLANE_GENERATED_FILE=${OUTPUT_DIR}/controlplane.yaml
+
+# Overwrite flag.
+OVERWRITE=0
+
+SCRIPT=$(basename "$0")
+while test $# -gt 0; do
+        case "$1" in
+          -h|--help)
+            echo "$SCRIPT - generates input yaml files for Cluster API on aws"
+            echo " "
+            echo "$SCRIPT [options]"
+            echo " "
+            echo "options:"
+            echo "-h, --help                show brief help"
+            echo "-f, --force-overwrite     if file to be generated already exists, force script to overwrite it"
+            exit 0
+            ;;
+          -f)
+            OVERWRITE=1
+            shift
+            ;;
+          --force-overwrite)
+            OVERWRITE=1
+            shift
+            ;;
+          *)
+            break
+            ;;
+        esac
+done
+
+if [ $OVERWRITE -ne 1 ] && [ -d "$OUTPUT_DIR" ]; then
+  echo "ERR: Folder ${OUTPUT_DIR} already exists. Delete it manually before running this script."
+  exit 1
+fi
+
+mkdir -p "${OUTPUT_DIR}"
+
+# Generate cluster resources.
+kustomize build "${SOURCE_DIR}/cluster" | envsubst > "${CLUSTER_GENERATED_FILE}"
+echo "Generated ${CLUSTER_GENERATED_FILE}"
+
+# Generate controlplane resources.
+kustomize build "${SOURCE_DIR}/controlplane" | envsubst > "${CONTROLPLANE_GENERATED_FILE}"
+echo "Generated ${CONTROLPLANE_GENERATED_FILE}"
+
+# Generate Cluster API provider components file.
+kustomize build "github.com/kubernetes-sigs/cluster-api//config/default/?ref=master" > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
```

Comment:
Is the extra slash in `cluster-api//config` a typo?

Also, I don't think `ref=master` is going to be viable in the long term.

Reply:
Not a typo 😄 It's to differentiate between the repository and the path

Reply:
Once we release a version, we can reference the tag directly instead of `master`